remix/app/routes/_app._index.test.tsx

```diff
@@ -0,0 +1,43 @@
+/**
+ * @jest-environment node
+ */
+
+import { loader } from './_app._index';
+import { authenticator } from '../config/auth.server';
+import { prismaMock } from '../tests/database';
+import { newsletterFactory } from '../tests/factories/newsletter.factory';
+import { userFactory } from '../tests/factories/user.factory';
+import { makeRequest } from '../tests/helpers/request';
+
+jest.mock('../config/auth.server', () => ({
+  authenticator: {
+    isAuthenticated: jest.fn(),
+  },
+}));
+
+describe('Root Page', () => {
+  describe('loader', () => {
+    it('returns a list of newsletters', async () => {
+      const userAtttributes = { id: '1' };
+      const newsletterAttributes = { userId: '1' };
+
+      const user = { ...userFactory, ...userAtttributes };
+      const newsletter = { ...newsletterFactory, ...newsletterAttributes };
+
+      const newsletters = Array(5).fill(newsletter);
+
+      (authenticator.isAuthenticated as jest.Mock).mockResolvedValue(user);
+      prismaMock.user.findUnique.mockResolvedValue(user);
```

Comment:
I'm curious why do we need to mock user? 👀

Reply:
because this `loader` uses `appHandler` which requires `userId` from the database.

Reply:
Ahh, I see. Yes, that will work too!

Usually, I prefer mocking the `appHandler` instead, since it's easier for the reviewer to review. As just looking into the 4 changed files of this PR, it's possible that the reviewer may forget about the implementation of the `appHandler`, hence, may be a bit confused here.

But this is just a minor so nvm 🙈

Reply:
Btw do you have any example of that ?

Reply:
Yes it is!

Reply:
Improved here: [6f2b9ad](https://github.com/nimblehq/js-framework-benchmark/pull/122/commits/6f2b9adad328a6e1eaacd9d4312a58be830a9e2d)

Reply:
LGTM 👍